README.md

```diff
@@ -1 +1,460 @@
-# golangcodestyle
\ No newline at end of file
+# Backend (Golang) Coding Guidelines
```

Comment:
5. Do not export functions and variables unless necessary. 
6. Keep the exported variables and error in the same package that is using/returning them. 
For example i/o error like `io.EOF` from the go std lib is in the `io` package, where the function returning this error are also defined. 

7. If your function return a specific error, and the caller might act differently if it receives this specific error, then document this behaviour. 

8. Do not return an unique error (an package level error variable that is exported) unless you are sure that the caller might actually need to discern between such error and any other error. 

9. Consider 
https://dave.cheney.net/2016/04/27/dont-just-check-errors-handle-them-gracefully
https://dave.cheney.net/2016/04/07/constant-errors
https://dave.cheney.net/2014/12/24/inspecting-errors (!)

Reply:
added: 
- https://github.com/apaliavy/golangcodestyle/blob/1eda2a9506765ca11274f4bf973baf16a84174af/README.md#exports
- https://github.com/apaliavy/golangcodestyle/blob/1eda2a9506765ca11274f4bf973baf16a84174af/README.md#errors